tests/testlib/s2n_test_certs.c

```diff
@@ -76,3 +76,33 @@ int s2n_read_test_pem_and_len(const char *pem_path, uint8_t *pem_out, uint32_t *
     return 0;
 }
 
+uint32_t write_pem_file_to_stuffer_as_chain(struct s2n_stuffer *chain_out_stuffer, const char *pem_data,
```

Comment:
Refactoring this was kind of a big change, so I moved it to its own PR: https://github.com/aws/s2n-tls/pull/3553/files

My understanding is the certificate message parsing is converting from DER, whereas this is converting from PEM. So I think the logic is different enough that reusing doesn't make sense. But if I'm missing something please let me know.

Reply:
No I meant that what it looks like what you're trying to produce is a stuffer containing a Certificate message. So could you parse the pem input the same way we do when setting certificates on a config, then use that config+connection to write a Certificate message? I'm not sure how much duplicate code it would save, but it seems odd to have the bit handling whether or not to write extensions in the current method.

Reply:
Ah, sorry. That definitely makes sense to do! I updated `s2n_test_cert_chain_from_pem` in that PR to reuse `s2n_send_cert_chain`.